### PR TITLE
Updating U5 challenges to remove no-args constructor

### PR DESCRIPTION
**Pull request by bhoffman0**
Mostly editing the challenge exercises in Unit 5 to remove the requirement to add a no-args constructor and changing the unit tests accordingly. I also added subsection headers to 5.2 and fixed an underline in 2.5 and removed some visualizer links.

### CODE REVIEW

File: _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst [hunk 1]
```diff
@@ -294,13 +301,11 @@ values themselves.
...
-   for example a Honda 2010 car. However, some of the code is missing. Fill in
-   the code for the 2 constructors that are numbered 1 and 2. And fill in the
-   code to call the constructors in the main method numbered 3. The car1 object
-   should test the first constructor with default values and the car2 object
-   should test the second constructor to create a Honda 2010 car. Run your
-   program and make sure it works and prints out the information for both cars.
-
+   for example a Honda 2010 car. However, some of the code is missing. First, fill in
+   the code to create a Car constructor. Then, fill in the code to call the constructor 
+   in the main method to create 2 Car objects. The first car should be a 2023 Ford and
```
Comment (gigamonkey) [inline _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst:306]:
```suggestion
   the code to create a ``Car`` constructor. Then, fill in the code to call the constructor 
   in the main method to create 2 ``Car`` objects. The first car should be a 2023 Ford and 
```

File: _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst [hunk 2]
```diff
@@ -635,7 +641,7 @@ Summary
 
 - **Constructors** are used to set the initial state of an object, which includes initial values for all instance variables.
 
-- When no constructor is written, Java provides a no-argument **default constructor**, and the instance variables are set to their default values (0 for int and double, null for objects like String).
+- When no constructor is written, Java provides a no-argument **default constructor**, and the instance variables are set to their default values (0 for ``int`` and ``double``, null for objects like ``String``).
```
Comment (gigamonkey) [inline _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst:644]:
```suggestion
- When no constructor is written, Java provides a no-argument **default constructor**, and the instance variables are set to their default values (0 for ``int`` and ``double``, ``null`` for objects like ``String``).
```

Comment (gigamonkey) [reply _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst:644]: I'm torn about how to format literals like `0`. Should they be formatted as code? In this case, I'd probably use the word "zero" except that the book as a whole seems to rather aggressively prefer numerals over words.

**Review by gigamonkey — APPROVED** [review]
I didn't actually run the new code but this all looks very reasonable to me. A couple suggested edits to add yet more missing backticks.

File: _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst (hunk 2 above)
Comment (bhoffman0) [reply _sources/Unit5-Writing-Classes/topic-5-2-writing-constructors.rst:644]: I don't think I would format the literal 0 as code, but either way works. I think I do have a preference for 0 over zero here, but not sure why!

**Pull request merged**